src/main/java/edu/cornell/kfs/fp/batch/service/impl/CuDisbursementVoucherDocumentGenerator.java

```diff
@@ -147,6 +150,51 @@ protected void validateAndPopulatePayeeTypeCodeAndPayeeNumber(DisbursementVouche
         payeeDetail.setDisbVchrPayeeIdNumber(payeeId);
     }
     
+    private void populateAddressFields(DisbursementVoucherPaymentInformationXml paymentInfo, CuDisbursementVoucherDocument dvDocument) {
+        if (StringUtils.isNotBlank(paymentInfo.getPayeeAddressId())) {
+            VendorAddress vendorAddress = findVendorAddress(paymentInfo);
+            dvDocument.templateVendor(vendorAddress.getVendorDetail(), vendorAddress);
+        } else {
+            CuDisbursementVoucherPayeeDetail payeeDetail = dvDocument.getDvPayeeDetail();
+            LOG.info("populateAddressFields, no payee address ID provided, so using the address fields from the XML");
+            payeeDetail.setDisbVchrPayeeLine1Addr(paymentInfo.getAddressLine1());
+            payeeDetail.setDisbVchrPayeeLine2Addr(paymentInfo.getAddressLine2());
+            payeeDetail.setDisbVchrPayeeCityName(paymentInfo.getCity());
+            payeeDetail.setDisbVchrPayeeStateCode(paymentInfo.getState());
+            payeeDetail.setDisbVchrPayeeCountryCode(paymentInfo.getCountry());
+            payeeDetail.setDisbVchrPayeeZipCode(paymentInfo.getPostalCode());
+        }
+    }
+
+    private VendorAddress findVendorAddress(DisbursementVoucherPaymentInformationXml paymentInfo) {
+        Collection<VendorAddress> vendorAddresses;
+        try {
+            String[] payeeArray = StringUtils.split(paymentInfo.getPayeeId(), KFSConstants.DASH);
+            if (LOG.isInfoEnabled()) {
+                LOG.info("findVendorAddress, vendorHeaderGeneratedIdentifier: " + payeeArray[0] + 
+                        " vendorDetailAssignedIdentifier: " + payeeArray[1] + 
+                        ", payee address id: " + paymentInfo.getPayeeAddressId());
+            }
+            Map<String, String> fieldValues = new HashMap<String, String>();
+            fieldValues.put(KFSPropertyConstants.VENDOR_HEADER_GENERATED_ID, payeeArray[0]);
+            fieldValues.put(KFSPropertyConstants.VENDOR_DETAIL_ASSIGNED_ID, payeeArray[1]);
+            fieldValues.put(KFSPropertyConstants.VENDOR_ADDRESS_GENERATED_ID, paymentInfo.getPayeeAddressId());
+            fieldValues.put(KFSPropertyConstants.ACCOUNT_ACTIVE_INDICATOR,  KFSConstants.ACTIVE_INDICATOR);
+            vendorAddresses = businessObjectService.findMatching(VendorAddress.class, fieldValues);
+        } catch (Throwable e) {
```

Comment:
Is catching a Throwable going to be too broad? Should this just capture Exception instead?

Reply:
I wanted to make sure every odd thing got caught here, and got handled ..... I was concerned with making sure any string split errors were handled

Reply:
I would think the string-splitting errors would extend from Exception. If you're catching at the level of Throwable, the code could end up catching out-of-memory errors and things like that.

Reply:
OK, I will switch to exception